src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/CollectMultipleMetricsSpark.java

```diff
@@ -0,0 +1,181 @@
+import org.broadinstitute.hellbender.metrics.QualityYieldMetricsArgs;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.util.*;
+import java.util.function.Consumer;
+
+/**
+ * Tool that instantiates and executes multiple metrics programs using a single RDD.
+ */
+@CommandLineProgramProperties(
+        summary = "Takes an input SAM/BAM/CRAM file and reference sequence and runs one or more " +
+                "metrics modules at the same time to cut down on I/O. Currently all programs are run with " +
+                "default options and fixed output extensions, but this may become more flexible in future.",
+        oneLineSummary = "A \"meta-metrics\" calculating program that produces multiple metrics for the provided SAM/BAM/CRAM file",
+        programGroup = SparkProgramGroup.class
+)
+public final class CollectMultipleMetricsSpark extends GATKSparkTool {
+
+    private static final long serialVersionUID = 1L;
+
+    @Argument(fullName=StandardArgumentDefinitions.ASSUME_SORTED_LONG_NAME,
+                shortName = StandardArgumentDefinitions.ASSUME_SORTED_SHORT_NAME,
+                doc = "If true (default), then the sort order in the header file will be ignored.")
+    public boolean ASSUME_SORTED = true;
+
+    //TODO: do we need this for Spark ?
+    //@Argument(doc = "Stop after processing N reads, mainly for debugging.")
+    //public int STOP_AFTER = 0;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+                shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+                doc = "Base name of output files.")
+    public String outputBaseName;
+
+    @Argument(shortName="LEVEL", doc="The level(s) at which to accumulate metrics.  ")
+    public Set<MetricAccumulationLevel> metricAccumulationLevel = EnumSet.of(MetricAccumulationLevel.ALL_READS);
+
+    @Argument(doc = "List of metrics collectors to apply during the pass through the SAM file.")
+    public List<Collectors> collectors = new ArrayList<>();
+
+    private interface CollectorPrimer {
+        /**
+         * For each collector type supported by this tool, we need to provide a
+         * type-safe, collector-specific primer that creates and populates an
+         * instance of the collector's input arguments class; creates an instance
+         * of the collector; initializes it with the arguments object; and then
+         * forwards the initialized collector to the executeLifeCycle function
+         */
+        void primeCollector(
+                final String outputBaseName,
+                final Set<MetricAccumulationLevel> metricAccumulationLevel,
+                final List<Header> defaultHeaders,
+                final Consumer<MetricsCollectorSpark<?>> executeCollectorLifecyle
+        );
+    }
+
+    // Enum of Collector types supported by this tool.
+    public static enum Collectors implements CollectorPrimer {
```

Comment:
I'm not a big fan of this hardcoded set of metrics.  

It seems like this is mostly just setting the metrics collectors to use what could be default values.  I.e. `InsertSizeMetricsCollectorSpark` should probably know to use '.insertMetrics' as it's default extension rather than having to define that here.  Maybe `MetricArgs` could provide a function to inject a base filepath?

Could `isArgs.useEnd` just default to SECOND?  

Maybe custom configuration of some collectors is necessary, but it seems like it could be avoided in these cases with some fairly limited extensions of the abstractions.